warp10/src/main/java/io/warp10/script/processing/typography/PcreateFont.java

```diff
@@ -31,9 +39,36 @@
  * Call createFont
  */
 public class PcreateFont extends NamedWarpScriptFunction implements WarpScriptStackFunction {
+    
+  private final FontResolver resolver;
   
+  private static class FontResolver {
+    
+    private final String warpscript;
+    
+    public FontResolver(String mc2) {
+      this.warpscript = mc2;
+    }
+    
+    public String resolve(String url) throws WarpScriptException {
+      if (null == this.warpscript) {
+        throw new WarpScriptException("Font resolver not set, rejecting all URLs.");
+      }
+      try {
+        MemoryWarpScriptStack stack = new MemoryWarpScriptStack(null, null);
+        stack.push(url);
+        stack.exec(this.warpscript);
+        return (String) stack.pop();
```

Comment:
Why use a resolver instead of a validator?
If it needs to stay a resolver, can't this macro return a special value for forbidden access, like `NULL` or an empty stack and produce a clear error message?

Reply:
Simply do a FAILMSG in your resolver if you want to reject the font. The WarpScriptException will be forwarded upstream.